freebites/product
# 67 admin view

# Description

We were able to both create the schema and backend functionality to store reports and were able to create the admin view to view all the reports.

## Issues

Resolve #67

## Screenshots


<img src="https://github.com/user-attachments/assets/fc951fa0-788a-49b6-ba25-37a73f5d58d4" width=200>
<img src="https://github.com/user-attachments/assets/2ac9275c-0c2e-4bfd-b5c9-d35ef702a3b3" width=200>
<img src="https://github.com/user-attachments/assets/a1e721b5-4d8f-449e-ac4d-950d0431662b" width=200>
<img src="https://github.com/user-attachments/assets/ab592174-411a-424c-b2ae-86ac1192aa84" width=200>
<img src="https://github.com/user-attachments/assets/4e433e1c-529b-494b-aeec-933827f8fceb" width=200>
<img src="https://github.com/user-attachments/assets/4bbc8630-2122-4588-9e99-a920619f830a" width=200>

## Test

Admin View:
- Log into an account that has admin access (the account we created was owen@mail.com with the same password as test@test.com)
- There should be an "All posts" tab and a "Report" tab
- Under each post under the "All posts" tab, there should be an "Achive" button that deletes the post and a report button that adds a report to the post
- Try reporting a post and see if the report shows up when clicking the "Report" tab.
- Under the "Report" tab, there should be a list of dropdowns that list all the usernames of the people who reported.
- Admin users can press on each report to see the report message and post that is associated with each report.
- The admin user can either ignore the report, archive the post, or add other reports to it



Currently, the only way turning an account into an admin account is if you manually go into the MongoDB Atlas and change their "isAdmin" field to "true".
